src/pudl/transform/vcerare.py

```diff
@@ -0,0 +1,330 @@
+    """
+    logger.info(f"Stacking the county/subregion columns for {df_name} table.")
+    df_stacked = (
+        df.set_index(["datetime_utc", "hour_of_year", "report_year"])
+        .stack()
+        .reset_index()
+        .rename(
+            columns={"level_3": "county_state_names", 0: f"capacity_factor_{df_name}"}
+        )
+        .assign(county_state_names=lambda x: pd.Categorical(x.county_state_names))
+    )
+    return df_stacked
+
+
+def _make_cap_fac_frac(df: pd.DataFrame, df_name: str) -> pd.DataFrame:
+    """Make the capacity factor column a fraction instead of a percentage.
+
+    This step happens before the table gets stacked to save memory.
+    """
+    logger.info(f"Converting capacity factor into a fraction for {df_name} table.")
+    county_cols = [x for x in df.columns if x not in ["report_year", "unnamed_0"]]
+    df[county_cols] = df[county_cols] / 100
+    return df
+
+
+def _check_for_valid_counties(
+    df: pd.DataFrame, clean_fips_df: pd.DataFrame, df_name: str
+) -> pd.DataFrame:
+    """Make sure the state_county values show up in the fips table.
+
+    This step happens before the table gets stacked to save memory.
+    """
+    logger.info(f"Checking for valid counties in the {df_name} table.")
+    county_state_names_fips = clean_fips_df.county_state_names.unique().tolist()
+    county_state_names_cap_fac = df.columns.tolist()
+    expected_non_counties = ["report_year", "unnamed_0"]
+    non_county_cols = [
+        x
+        for x in county_state_names_fips
+        if x not in county_state_names_cap_fac + expected_non_counties
+    ]
+    if non_county_cols:
+        raise AssertionError(f"""found unexpected columns that aren't in the FIPS table:
+{non_county_cols}.""")
+    return df
+
+
+def _combine_all_cap_fac_dfs(cap_fac_dict: dict[str, pd.DataFrame]) -> pd.DataFrame:
+    """Combine capacity factor tables."""
+    logger.info("Merging all the capacity factor tables into one")
+    merge_keys = ["report_year", "datetime_utc", "hour_of_year", "county_state_names"]
+    for name, df in cap_fac_dict.items():
+        cap_fac_dict[name] = df.set_index(merge_keys)
+    mega_df = pd.concat(
+        [
+            cap_fac_dict["solar_pv"],
+            cap_fac_dict["offshore_wind"],
+            cap_fac_dict["onshore_wind"],
+        ],
+        axis="columns",
+    ).reset_index()
+
+    return mega_df
+
+
+def _combine_cap_fac_with_fips_df(
+    cap_fac_df: pd.DataFrame, fips_df: pd.DataFrame
+) -> pd.DataFrame:
+    """Combine the combined capacity factor df with the fips df."""
+    logger.info(
+        "Merging the combined capacity factor table with the Lat-Long-FIPS table"
+    )
+    combined_df = pd.merge(cap_fac_df, fips_df, on="county_state_names", how="left")
+    return combined_df
+
+
+def _combine_city_county_records(df: pd.DataFrame) -> pd.DataFrame:
+    """Average cap fac values for city and county values with the same FIPS ID.
```

Comment:
Personally think it's almost neater to just drop the city value, but you've gone through the work of averaging it so I think we can just keep it. It would be helpful to add an assertion that ensures that this is the only non-lake case with a duplicated county and values.

Reply:
Looking at [Google Maps](https://www.google.com/maps/place/Alleghany+County,+VA/@37.8614873,-80.0798489,87885m/data=!3m1!1e3!4m6!3m5!1s0x884cc397910e13fd:0xc43980e08ca7e1d0!8m2!3d37.8531975!4d-80.0534923!16zL20vMG1wZGg?hl=en&entry=ttu&g_ep=EgoyMDI0MTAxNS4wIKXMDSoASAFQAw%3D%3D) Clifton Forge appears to be quite a small town in Alleghany County, VA so dropping it rather than having it influence the overall generation profile for the county seems like it would be the bbest option.

Reply:
Ok easy enough -- I changed the function so that it drops these two cities.